src/Polynomials/MonomialBases.jl

```diff
@@ -406,6 +475,35 @@ function _gradient_nd!(
 
 end
 
+function _gradient_nd!(
```

Comment:
While it is certainly a possibility to dispatch also on Tisbitstype at the `_gradient_nd!` level, I dont think it is actually needed. You can move `z = zero(Mutable(VectorValue{D,T}))` to the caller function, and then you do not need two additional function definitions for `_gradient_nd!`. Agreed?

Reply:
by the caller function you mean `_evaluate` ?

Reply:
The function that calls `_gradient_nd`.

Reply:
done